src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1043,46 +1041,70 @@ private enum AssignmentKind
             Conversion conversion = default,
             Location location = null)
         {
+            // PROTOTYPE: Fix callers so we don't have cases where valueType and targetType differ.
```

Comment:
In which scenarios do they differ?

Reply:
The types may differ in binary logical operators and deconstruction. See [https://github.com/dotnet/roslyn/issues/39867](https://github.com/dotnet/roslyn/issues/39867).

---
In reply to: [347048526](https://github.com/dotnet/roslyn/pull/39778#discussion_r347048526) [](ancestors = 347048526)